Compile geos error

I have downloaded GEOS(develop), and installed the tools according to the build guide. The environments are as follow below:
GEOS version:develop
os: WSL Ubuntu 24.04.1 LTS
gcc (Ubuntu 13.3.0-6ubuntu224.04) 13.3.0
g++ (Ubuntu 13.3.0-6ubuntu224.04) 13.3.0
GNU Fortran (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
cmake version 3.28.3
GNU Make 4.3

The TPLs have compiled successfully, and I encounters some errors when compiling GEOS. Can anyone give some advice?

```
In file included from /mnt/d/codes/GEOSX/src/coreComponents/LvArray/src/ArrayView.hpp:21,
                 from /mnt/d/codes/GEOSX/src/coreComponents/LvArray/src/Array.hpp:17,
                 from /mnt/d/codes/GEOSX/src/coreComponents/common/DataLayouts.hpp:25,
                 from /mnt/d/codes/GEOSX/src/coreComponents/common/DataTypes.hpp:30,
                 from /mnt/d/codes/GEOSX/src/coreComponents/common/format/StringUtilities.hpp:23,
                 from /mnt/d/codes/GEOSX/src/coreComponents/codingUtilities/Utilities.hpp:23,
                 from /mnt/d/codes/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.hpp:23,
                 from /mnt/d/codes/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp:20:
In function ‘void LvArray::bufferManipulation::copyInto(DST_BUFFER&, std::ptrdiff_t, const SRC_BUFFER&, std::ptrdiff_t) [with DST_BUFFER = LvArray::StackBuffer<double, 9>; SRC_BUFFER = LvArray::StackBuffer<double, 9>]’,
    inlined from ‘LvArray::Array<T, NDIM, PERMUTATION, INDEX_TYPE, BUFFER_TYPE>& LvArray::Array<T, NDIM, PERMUTATION, INDEX_TYPE, BUFFER_TYPE>::operator=(const LvArray::Array<T, NDIM, PERMUTATION, INDEX_TYPE, BUFFER_TYPE>&) [with T = double; int NDIM = 1; PERMUTATION = camp::int_seq<long int, 0>; INDEX_TYPE = int; BUFFER_TYPE = LvArray::internal::StackArrayHelper<double, 1, camp::int_seq<long int, 0>, int, 9>::BufferType]’ at /mnt/d/codes/GEOSX/src/coreComponents/LvArray/src/Array.hpp:175:33,
    inlined from ‘LvArray::Array<T, NDIM, PERMUTATION, INDEX_TYPE, BUFFER_TYPE>::Array(const LvArray::Array<T, NDIM, PERMUTATION, INDEX_TYPE, BUFFER_TYPE>&) [with T = double; int NDIM = 1; PERMUTATION = camp::int_seq<long int, 0>; INDEX_TYPE = int; BUFFER_TYPE = LvArray::internal::StackArrayHelper<double, 1, camp::int_seq<long int, 0>, int, 9>::BufferType]’ at /mnt/d/codes/GEOSX/src/coreComponents/LvArray/src/Array.hpp:140:11,
    inlined from ‘static void geos::compositionalMultiphaseWellKernels::PresTempCompFracInitializationKernel::launch(geos::localIndex, geos::localIndex, geos::integer, geos::integer, geos::localIndex, const geos::WellControls&, const geos::real64&, ElementViewConst<LvArray::ArrayView<const double, 1, 0, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 1, 0, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 2, 1, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 2, 1, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 3, 2, int, LvArray::ChaiBuffer> >&, geos::arrayView1d<const int>&, geos::arrayView1d<const int>&, geos::arrayView1d<const int>&, geos::arrayView1d<const double>&, geos::arrayView1d<const double>&, geos::arrayView1d<double>&, geos::arrayView1d<double>&, geos::arrayView2d<double>&)’ at /mnt/d/codes/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp:636:52:
/mnt/d/codes/GEOSX/src/coreComponents/LvArray/src/bufferManipulation.hpp:409:18: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ forming offset [80, 87] is out of the bounds [0, 80] of object ‘avgCompFrac’ with type ‘geos::stackArray1d<double, 9>’ {aka ‘LvArray::Array<double, 1, camp::int_seq<long int, 0>, int, LvArray::internal::StackArrayHelper<double, 1, camp::int_seq<long int, 0>, int, 9>::BufferType>’} [-Werror=array-bounds=]
  409 |     dstData[ i ] = srcData[ i ];
      |     ~~~~~~~~~~~~~^~~~~~~~~~
/mnt/d/codes/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp: In static member function ‘static void geos::compositionalMultiphaseWellKernels::PresTempCompFracInitializationKernel::launch(geos::localIndex, geos::localIndex, geos::integer, geos::integer, geos::localIndex, const geos::WellControls&, const geos::real64&, ElementViewConst<LvArray::ArrayView<const double, 1, 0, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 1, 0, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 2, 1, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 2, 1, int, LvArray::ChaiBuffer> >&, ElementViewConst<LvArray::ArrayView<const double, 3, 2, int, LvArray::ChaiBuffer> >&, geos::arrayView1d<const int>&, geos::arrayView1d<const int>&, geos::arrayView1d<const int>&, geos::arrayView1d<const double>&, geos::arrayView1d<const double>&, geos::arrayView1d<double>&, geos::arrayView1d<double>&, geos::arrayView2d<double>&)’:
/mnt/d/codes/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp:561:40: note: ‘avgCompFrac’ declared here
  561 |   stackArray1d< real64, MAX_NUM_COMP > avgCompFrac( numComps );
      |                                        ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/build.make:412: coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

I got the same error. TPLs compiled successfully.

After the message posted by @liangfangzhao , the outputs go a little more, stopping at this line (64%):

```
[ 64%] Linking CXX shared library ../../../lib/libsolidMechanicsSolvers.so
[ 64%] Built target solidMechanicsSolvers
make: *** [Makefile:146: all] Error 2
```

My environment is 

```
OS: Windows Subsystem for Linux (WSL2) - Ubuntu 24.04.2 LTS - Kernel: Linux 5.15.167.4-microsoft-standard-WSL2
cpp (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
cmake version 3.28.3
GNU Fortran (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
GNU Make 4.3
```

--- a/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp
+++ b/src/coreComponents/physicsSolvers/fluidFlow/wells/kernels/CompositionalMultiphaseWellKernels.cpp
@@ -632,6 +632,7 @@ PresTempCompFracInitializationKernel::
   RAJA::ReduceMax< parallelDeviceReduce, integer > foundNegativePres( 0 );
   RAJA::ReduceMax< parallelDeviceReduce, integer > foundInconsistentCompFrac( 0 );
 
+  auto const avgCompFracView = avgCompFrac.toViewConst();
 
   forAll< parallelDevicePolicy<> >( subRegionSize, [=] GEOS_HOST_DEVICE ( localIndex const iwelem )
   {
@@ -641,7 +642,7 @@ PresTempCompFracInitializationKernel::
     real64 sumCompFracForCheck = 0.0;
     for( integer ic = 0; ic < numComps; ++ic )
     {
-      wellElemCompFrac[iwelem][ic] = avgCompFrac[ic];
+      wellElemCompFrac[iwelem][ic] = avgCompFracView[ic];
       sumCompFracForCheck += wellElemCompFrac[iwelem][ic];
     }
 

--- a/src/coreComponents/unitTests/constitutiveTests/MultiFluidTest.hpp
+++ b/src/coreComponents/unitTests/constitutiveTests/MultiFluidTest.hpp
@@ -225,13 +225,13 @@ testNumericalDerivatives( constitutive::MultiFluidBase & fluid,
     GET_FLUID_DATA( fluid, fields::multifluid::dTotalDensity )
   };
 
-  auto phaseFracCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseFraction );
-  auto phaseDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseDensity );
-  auto phaseViscCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseViscosity );
-  auto phaseEnthCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseEnthalpy );
-  auto phaseEnergyCopy   = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseInternalEnergy );
-  auto phaseCompFracCopy = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseCompFraction );
-  auto totalDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::totalDensity );
+  auto const & phaseFracCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseFraction );
+  auto const & phaseDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseDensity );
+  auto const & phaseViscCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseViscosity );
+  auto const & phaseEnthCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseEnthalpy );
+  auto const & phaseEnergyCopy   = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseInternalEnergy );
+  auto const & phaseCompFracCopy = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseCompFraction );
+  real64 totalDensCopy = 0.0;
 
 #undef GET_FLUID_DATA
 
@@ -282,6 +282,8 @@ testNumericalDerivatives( constitutive::MultiFluidBase & fluid,
       resetFluid( fluidCopy );
       fluidWrapper.update( 0, 0, pressure + dP, temperature, composition );
 
+      totalDensCopy = fluidCopy.totalDensity()( 0, 0 );
+
       checkDerivative( phaseFracCopy.toSliceConst(), phaseFrac.value.toSliceConst(), dPhaseFrac[Deriv::dP].toSliceConst(),
                        dP, relTol, absTol, "phaseFrac", "Pres", phases );
       checkDerivative( phaseDensCopy.toSliceConst(), phaseDens.value.toSliceConst(), dPhaseDens[Deriv::dP].toSliceConst(),
@@ -308,6 +310,8 @@ testNumericalDerivatives( constitutive::MultiFluidBase & fluid,
       real64 const dT = perturbParameter * (temperature + perturbParameter);
       resetFluid( fluidCopy );
       fluidWrapper.update( 0, 0, pressure, temperature + dT, composition );
+
+      totalDensCopy = fluidCopy.totalDensity()( 0, 0 );
 
       checkDerivative( phaseFracCopy.toSliceConst(), phaseFrac.value.toSliceConst(), dPhaseFrac[Deriv::dT].toSliceConst(),
                        dT, relTol, absTol, "phaseFrac", "Temp", phases );
@@ -366,6 +370,8 @@ testNumericalDerivatives( constitutive::MultiFluidBase & fluid,
 
       resetFluid( fluidCopy );
       fluidWrapper.update( 0, 0, pressure, temperature, compNew[0] );
+
+      totalDensCopy = fluidCopy.totalDensity()( 0, 0 );
 
       string const var = "compFrac[" + components[jc] + "]";
       checkDerivative( phaseFracCopy.toSliceConst(), phaseFrac.value.toSliceConst(), dPhaseFrac[Deriv::dC+jc].toSliceConst(),

--- a/src/coreComponents/unitTests/constitutiveTests/MultiFluidTest.hpp
+++ b/src/coreComponents/unitTests/constitutiveTests/MultiFluidTest.hpp
@@ -225,13 +225,13 @@ testNumericalDerivatives( constitutive::MultiFluidBase & fluid,
     GET_FLUID_DATA( fluid, fields::multifluid::dTotalDensity )
   };
 
-  auto const phaseFracCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseFraction );
-  auto const phaseDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseDensity );
-  auto const phaseViscCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseViscosity );
-  auto const phaseEnthCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseEnthalpy );
-  auto const phaseEnergyCopy   = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseInternalEnergy );
-  auto const phaseCompFracCopy = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseCompFraction );
-  auto const totalDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::totalDensity );
+  auto phaseFracCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseFraction );
+  auto phaseDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseDensity );
+  auto phaseViscCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseViscosity );
+  auto phaseEnthCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseEnthalpy );
+  auto phaseEnergyCopy   = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseInternalEnergy );
+  auto phaseCompFracCopy = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseCompFraction );
+  auto totalDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::totalDensity );
 
 #undef GET_FLUID_DATA
 

--- a/src/coreComponents/unitTests/constitutiveTests/MultiFluidTest.hpp
+++ b/src/coreComponents/unitTests/constitutiveTests/MultiFluidTest.hpp
@@ -225,13 +225,13 @@ testNumericalDerivatives( constitutive::MultiFluidBase & fluid,
     GET_FLUID_DATA( fluid, fields::multifluid::dTotalDensity )
   };
 
-  auto const & phaseFracCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseFraction );
-  auto const & phaseDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseDensity );
-  auto const & phaseViscCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseViscosity );
-  auto const & phaseEnthCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseEnthalpy );
-  auto const & phaseEnergyCopy   = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseInternalEnergy );
-  auto const & phaseCompFracCopy = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseCompFraction );
-  auto const & totalDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::totalDensity );
+  auto const phaseFracCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseFraction );
+  auto const phaseDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseDensity );
+  auto const phaseViscCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseViscosity );
+  auto const phaseEnthCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseEnthalpy );
+  auto const phaseEnergyCopy   = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseInternalEnergy );
+  auto const phaseCompFracCopy = GET_FLUID_DATA( fluidCopy, fields::multifluid::phaseCompFraction );
+  auto const totalDensCopy     = GET_FLUID_DATA( fluidCopy, fields::multifluid::totalDensity );
 
 #undef GET_FLUID_DATA
 

